modules/tensor_mechanics/examples/piston/piston_params.i

```diff
@@ -0,0 +1,178 @@
+## This example is documented on YouTube at:
+## https://www.youtube.com/watch?v=L9plLYkAbvQ
+## 
+## Additional files (e.g. the CAD model, results)
+## can be downloaded freely from Zenodo at:
+## https://doi.org/10.5281/zenodo.3886965
+
+[Mesh]
+  displacements = 'disp_x disp_y disp_z' #Define displacements for deformed mesh
+  type = FileMesh #Read in mesh from file
+  file = piston_coarse.e
+[]
+
+# This is where mesh adaptivity magic happens:
+[./Adaptivity]
+  steps = 1
+  #initial_steps = 2
+  max_h_level = 3
+  cycles_per_step = 1
+  initial_marker = uniform
+  marker = errorFraction
+  [./Markers]
+    [./uniform]
+      type = UniformMarker
+      mark = refine
+    [../]
+    
+    [./errorFraction]
+      type = ErrorFractionMarker
+      coarsen = 0.5
+      indicator = gradientJump
+      refine = 0.5
+    [] 
+ [../]
+
+  [./Indicators]
+    [./gradientJump]
+      type = GradientJumpIndicator
+      variable = disp_y
+    []
+  []
+[../]
+
+[Variables]
+  [./disp_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Kernels]
+  [./TensorMechanics]
+    #Stress divergence kernels
+    displacements = 'disp_x disp_y disp_z'
+  [../]
+[]
+
+[AuxVariables]
+  [./von_mises]
+    #Dependent variable used to visualize the Von Mises stress
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[AuxKernels]
+  [./von_mises_kernel]
+    #Calculates the von mises stress and assigns it to von_mises
+    type = RankTwoScalarAux
+    variable = von_mises
+    rank_two_tensor = stress
+    execute_on = timestep_end
+    scalar_type = VonMisesStress
+  [../]
+[]
```

Comment:
With tensor mechanics action, you can use `generate_output = vonmises_stress` for outputting.

Reply:
See https://github.com/idaholab/moose/pull/18543/commits/b9fba72d5aea38f9559e6c654d576ce79737a490